src/swap/SwapTransactionDetails.tsx

```diff
@@ -194,15 +200,52 @@ export function SwapTransactionDetails({
         placeholder={placeholder}
         testID={`SwapTransactionDetails/MaxNetworkFee`}
       />
-      {/* When enabled, the fee is part of the exchange rate */}
-      {!enableAppFee && (
-        <View style={styles.row}>
-          <Text style={styles.label}>{t('swapScreen.transactionDetails.swapFee')}</Text>
-          <Text testID={'SwapFee'} style={styles.value}>
-            {t('swapScreen.transactionDetails.swapFeeWaived')}
-          </Text>
-        </View>
-      )}
+      <View style={styles.row}>
+        <LabelWithInfo
+          onPress={() => {
+            ValoraAnalytics.track(SwapEvents.swap_show_info, {
+              type: SwapShowInfoType.APP_FEE,
+            })
+            appFeeInfoBottomSheetRef.current?.snapToIndex(0)
+          }}
+          label={t('swapScreen.transactionDetails.appFee')}
+          testID="SwapTransactionDetails/AppFee/MoreInfo"
+        />
+        <Text testID={'SwapTransactionDetails/AppFee'} style={styles.value}>
+          <Trans
+            i18nKey={'swapScreen.transactionDetails.appFeeValue'}
+            context={
```

Comment:
this kind of feels too smart use for [context](https://www.i18next.com/translation-function/context) 😅

it implies this `context` prop conditions and underlying children rendering conditions must be kept in sync (e.g. both have to rely on `appFee.token.priceUsd`). i expect it to be hard/error-prone to quickly adjust in the future: one has to account for this, and the `translation.json` template as well.

wouldn't the good old conditional rendering without translation templates/context be less complex?

Reply:
Yes it is a bit complex. But we need the ability to properly translate this.

The children conditionals aren't entirely needed, I did that to make it easier to understand tests.

We could have 4 very similar renders. Probably easier to understand but with more duplication. Though there's also test coverage.

In the interest of time, I can take a stab at this later to not block the current release further than needed. What do you think?

Reply:
yes, I agree!
I'm not sure if we actually have to tune it given there is a refactoring of swap details is ahead